docs/verify/simulate-with-qiskit-primitives.mdx

```diff
@@ -269,189 +249,253 @@ qc.cx(0,1)
-
-```python
-job = sampler.run(qc)
-result = job.result()
-print(result)
-```
+  <TabItem value="SamplerV1" label="Sampler (V1)">
+    ```python
+from qiskit.primitives import Sampler
 
-```python
-SamplerResult(quasi_dists=[{0: 0.4999999999999999, 3: 0.4999999999999999}], metadata=[{}])
+sampler = Sampler()
 ```
+  </TabItem>
+</Tabs>
+### Run and get results
 
-While this example only uses one [`qiskit.circuit.QuantumCircuit`](../api/qiskit/qiskit.circuit.QuantumCircuit), you can sample multiple circuits by passing a `list` of [`qiskit.circuit.QuantumCircuit`](../api/qiskit/qiskit.circuit.QuantumCircuit) instances to the [`qiskit.primitives.Sampler.run`](../api/qiskit/qiskit.primitives.Sampler#run) method.
-
-### Get the probability distribution
-
-From these results you can extract the quasi-probability distributions with the attribute [`qiskit.primitives.SamplerResult.quasi_dists`](../api/qiskit/qiskit.primitives.SamplerResult#quasi_dists).
-
-Even though there is only one circuit in this example, [`qiskit.primitives.SamplerResult.quasi_dists`](../api/qiskit/qiskit.primitives.SamplerResult#quasi_dists) returns a list of [`qiskit.result.QuasiDistribution`](../api/qiskit/qiskit.result.QuasiDistribution)s.
-`result.quasi_dists[i]` is the quasi-probability distribution of the `i`<sup>th</sup> circuit.
+<Tabs>
+  <TabItem value="SamplerV2" label="StatevectorSampler (V2)">
+    ```python
+    # Circuits and parameters must obey the Instruction Set Architecture (ISA) of a particular backend.
+    from qiskit.transpiler.preset_passmanagers import generate_preset_pass_manager
+
+    pm = generate_preset_pass_manager(backend=backend, optimization_level=1)
+    isa_circuit = pm.run(circuit)
+
+    # execute 1 circuit with 3 parameter sets using Sampler V2
+    job = sampler.run([(isa_circuit, [vals1, vals2, vals3])])
+    counts = job.result()[0].data.meas.get_counts()
+    ```
+  V2 primitives accept multiple PUBs as inputs, and each PUB gets its own result.
+  Therefore, you can run different circuits with various parameter/observable combinations,
+  which was not always possible in the V1 interface:
+
+    ```python
+    # Circuits and parameters must obey the Instruction Set Architecture (ISA) of a particular backend.
+    from qiskit.transpiler.preset_passmanagers import generate_preset_pass_manager
+
+    pm = generate_preset_pass_manager(backend=backend, optimization_level=1)
+    isa_circuit1 = pm.run(circuit1)
+    isa_circuit2 = pm.run(circuit2)
+
+    # execute 2 circuits with 1 parameter set using Sampler V2
+    job = sampler.run([(isa_circuit1, vals1), (isa_circuit2, vals1)])
+    counts1 = job.result()
+    counts2 = job.result()
```

Comment:
There were some bugs in this snippet (probably my fault): there is no backend when working with reference primitives (so we can't do `generate_preset_pass_manager(backend=backend, optimization_level=1)`) and the result retrieval was not following the V2 syntax:

```suggestion
    from qiskit.transpiler.preset_passmanagers import generate_preset_pass_manager
    
    # create two circuits
    circuit1 = qc.copy()
    circuit2 = qc.copy()
    
    # transpile circuits 
    pm = generate_preset_pass_manager(optimization_level=1)
    isa_circuit1 = pm.run(circuit1)
    isa_circuit2 = pm.run(circuit2)

    # execute 2 circuits with 1 parameter set using Sampler V2
    job = sampler.run([(isa_circuit1, vals1), (isa_circuit2, vals1)])
    pub_result_1 = job.result()[0]
    pub_result_2 = job.result()[1]
```

I also removed the comment about ISA because we actually don't have any backend to refer to (I'm fine with still calling the transpiled circuits isa_circuit because I think it helps, but tecnically there is no ISA here).